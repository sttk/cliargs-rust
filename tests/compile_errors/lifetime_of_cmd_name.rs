use cliargs::Cmd;

fn main() {
    fn returns_cmd_name() -> &'static str {
        let mut cmd = Cmd::with_strings(["/path/to/app".to_string()]);
        cmd.parse().unwrap();

        let name = cmd.name();
        println!("command name (within the scope = {name}");

        name
    }

    let name = returns_cmd_name();
    println!("command name (out of the scope) = {name}");
}
