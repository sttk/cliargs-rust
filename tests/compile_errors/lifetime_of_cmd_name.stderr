error[E0515]: cannot return value referencing local variable `cmd`
  --> tests/compile_errors/lifetime_of_cmd_name.rs:11:9
   |
8  |         let name = cmd.name();
   |                    --- `cmd` is borrowed here
...
11 |         name
   |         ^^^^ returns a value referencing data owned by the current function
